File: Lib/Telerik/Examples/FileExplorer/CustomToggleButton.vb
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Primitives

Namespace FileExplorer
	Friend Class CustomToggleButton
		Inherits RadToggleButtonElement

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

		End Sub

		Protected Overrides Sub OnToggleStateChanged(ByVal e As StateChangedEventArgs)
			If e.ToggleState = Telerik.WinControls.Enumerations.ToggleState.On Then
                Me.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Visible
                Me.ButtonFillElement.ResetValue(FillPrimitive.BackColorProperty, Telerik.WinControls.ValueResetFlags.Local)
                Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor2Property, Telerik.WinControls.ValueResetFlags.Local)
                Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor3Property, Telerik.WinControls.ValueResetFlags.Local)
                Me.ButtonFillElement.ResetValue(FillPrimitive.BackColor4Property, Telerik.WinControls.ValueResetFlags.Local)
			Else
                Me.ButtonFillElement.BackColor = Color.Transparent
                Me.ButtonFillElement.BackColor2 = Color.Transparent
                Me.ButtonFillElement.BackColor3 = Color.Transparent
                Me.ButtonFillElement.BackColor4 = Color.Transparent
                Me.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Collapsed
			End If
			MyBase.OnToggleStateChanged(e)
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(RadToggleButtonElement)
			End Get
		End Property
	End Class
End Namespace

File: Lib/Telerik/Examples/FileExplorer/CustomToggleButton.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.WinControls.Primitives;
using System.Drawing;

namespace FileExplorer
{
    class CustomToggleButton : RadToggleButtonElement
    {

        protected override void CreateChildElements()
        {
            base.CreateChildElements();
         
        }

        protected override void OnToggleStateChanged(StateChangedEventArgs e)
        {
            if (e.ToggleState == Telerik.WinControls.Enumerations.ToggleState.On)
            {
                this.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Visible;
                this.ButtonFillElement.ResetValue(FillPrimitive.BackColorProperty, Telerik.WinControls.ValueResetFlags.Local);
                this.ButtonFillElement.ResetValue(FillPrimitive.BackColor2Property, Telerik.WinControls.ValueResetFlags.Local);
                this.ButtonFillElement.ResetValue(FillPrimitive.BackColor3Property, Telerik.WinControls.ValueResetFlags.Local);
                this.ButtonFillElement.ResetValue(FillPrimitive.BackColor4Property, Telerik.WinControls.ValueResetFlags.Local);
            }
            else
            {
                this.ButtonFillElement.BackColor = Color.Transparent;
                this.ButtonFillElement.BackColor2 = Color.Transparent;
                this.ButtonFillElement.BackColor3 = Color.Transparent;
                this.ButtonFillElement.BackColor4 = Color.Transparent;
                this.BorderElement.Visibility = Telerik.WinControls.ElementVisibility.Collapsed;
            }
            base.OnToggleStateChanged(e);
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadToggleButtonElement);
            }
        }
    }
}
